blog/2020-01-everything-github/README.md

```diff
@@ -0,0 +1,713 @@
+  ```
+   
+## 4. A first deployment to GitHub Pages
+
+We are now ready to host our first app on GitHub Pages.
+For this, the project must be compiled first and the compiled assets should be pushed to a new branch.
+That functionality is provided by `angular-cli-ghpages`.
+GitHub will activate the hosting automatically, if this branch has the name `gh-pages` – which is the default setting for `angular-cli-ghpages`.
+
+1. The following command adds `angular-cli-ghpages` to your project.
+
+    ```sh
+    ng add angular-cli-ghpages
+    ```
+
+2. Now we can deploy our project to GitHub pages with all default settings.
+    The project will be automatically built in production mode:
+
+    ```sh
+   ng deploy
+    ```
+
+    The deployment will create a temporary git repository and will commit/push all files 
+    from `dist/everything-github-demo` to a branch with the name `gh-pages`.
+    
+    But in this example we haven't provided a target repository.
+    In this case the deployment assumes that the current working directory is already a Git repository,
+    and that we want to push changes to the same repository with the name `origin`.
+    We have prepared this before with the command `git remote add`.  
+
+3. The app should be available at `https://<username>.github.io/<repositoryname>/` soon.
+   If it is not immediately available, you should first wait a moment.
+
+4. If we now check the results (in our case at https://angular-schule.github.io/everything-github-demo/) we will see a blank page! 😲
+    But no problem, we open the Console panel of the Chrome DevTools by pressing `Control+Shift+I` (Windows, Linux) or `Command+Option+I` (Mac).
+    We immediately see some red 404 errors for all files:
+
+    > Failed to load resource: the server responded with a status of 404 ()
+
+    Well, the reason for that is simple:
+    The application is configured as if it is running on the root path of the domain.
+    But this is not the case here!
+    Our application is located in the `/<repositoryname>/` path, so all links are broken.
+    This is of course adjustable.
+    Therefore, we will try the deployment a second time:
+
+    ```sh
+    ng deploy --base-href=/<repositoryname>/
+
+    # so in our case it must be
+    ng deploy --base-href=/everything-github-demo/
+    ```
+
+    **Hint:** Please make sure that you are really adding the trailing the slash.
+    If you forget it, it won't work!
+
+5. And now you should see our app running on Github Pages!
+    You can make sure that everything went well when you go into the **Settings** of the repo.
+    Scroll down there to the GitHub Pages section.
+    The following should appear here:
+
+    ![Screenshot](screenshot_4_settings.png)
+
+    **Hint:** You can also see the setting for a custom domain here.
+    Please do not try to set a value.
+    This will work for the first time, but the setting is not permanent, because `angular-cli-ghpages` will overwrite it again.
+
+
```

Comment:
Just to avoid any un-wanted errors for our readers, add a step to commit and push at the end of 4. A first deployment to GitHub Pages. Otherwise, their pipeline will fail. 😅
```suggestion
6. Now, add, commit and push the changes:
    ```sh
    git add .
    git commit -m "build: add angular-cli-gh-pages"
    git push origin master
    ```
```